qcfractal/storage_sockets/sqlalchemy_socket.py

```diff
@@ -1660,11 +1668,16 @@ def queue_mark_complete(self, task_ids: List[str]) -> int:
 
         update_fields = dict(status=TaskStatusEnum.complete, modified_on=dt.utcnow())
         with self.session_scope() as session:
-            # Update the base results
-            session.query(BaseResultORM)\
-                   .filter(BaseResultORM.id == TaskQueueORM.base_result_id)\
-                   .filter(TaskQueueORM.id.in_(task_ids))\
-                   .update(update_fields, synchronize_session=False)
+            for task_id in task_ids:
```

Comment:
Out of curiosity is this for loop slow? It should just template the values into a single SQL query correct?

Reply:
Surely it's slower than the original code. I've just found out the SQLAlchemy supports case statement in sql:)
I'll update that.
SQLAlchemy is great!

Reply:
Awesome :)